src/blockMonitor/blockProcessor.ts

```diff
@@ -171,8 +173,22 @@ export class BlockProcessor<TBlock extends IBlockStub> extends StartStopService
             this.mBlockCache.setHead(headBlock.hash);
 
             // only emit new head events after it is started
-            if (this.started) this.newHead.emit(headBlock);
+            if (this.started) {
+                try {
+                    await this.blockItemStoreLock.acquire();
+                    await this.blockItemStore.withBatch(
+                        // All the writes in the BlockItemStore that happen in any of the components are executed as part of the same batch.
+                        // Thus, they are either all written to the db, or none of them is.
```

Comment:
Again, I think you should include more detail in this comment.

Reply:
Comment expanded in 5828043.